clang/test/CheckedCRewriter/alloc_type_param.c

```diff
@@ -1,71 +1,78 @@
+// RUN: cconv-standalone -output-postfix=checked -alltypes %s
+// RUN: cconv-standalone -alltypes %S/alloc_type_param.checked.c -- | count 0
+// RUN: rm %S/alloc_type_param.checked.c
 
 typedef unsigned long size_t;
 extern _Itype_for_any(T) void *calloc(size_t nmemb, size_t size) : itype(_Array_ptr<T>) byte_count(nmemb * size);
 extern _Itype_for_any(T) void *malloc(size_t size) : itype(_Array_ptr<T>) byte_count(size);
 extern _Itype_for_any(T) void *realloc(void *pointer : itype(_Array_ptr<T>) byte_count(1), size_t size) : itype(_Array_ptr<T>) byte_count(size);
 
-// Check basic behavior with the three alloc functions
+/* Check basic behavior with the three alloc functions */
 void foo() {
   int *a = malloc(sizeof(int));
-  // CHECK: _Ptr<int> a = malloc<int>(sizeof(int));
+	//CHECK: _Ptr<int> a =  malloc<int>(sizeof(int));
   int *b = calloc(1, sizeof(int));
-  // CHECK: _Ptr<int> b = calloc<int>(1, sizeof(int));
+	//CHECK: _Ptr<int> b =  calloc<int>(1, sizeof(int));
   int *c = realloc(a, sizeof(int));
-  // CHECK: _Ptr<int> c = realloc<int>(a, sizeof(int));
+	//CHECK: _Ptr<int> c =  realloc<int>(a, sizeof(int));
 
-  // Explicit casts work fine also
+  /* Explicit casts work fine also */
 
   int *d = (int*) malloc(sizeof(int));
-  // CHECK: _Ptr<int> d = (_Ptr<int> ) malloc<int>(sizeof(int));
+	//CHECK: _Ptr<int> d =  (_Ptr<int> ) malloc<int>(sizeof(int));
   int *e = (int*) calloc(1, sizeof(int));
-  // CHECK: _Ptr<int> e = (_Ptr<int> ) calloc<int>(1, sizeof(int));
+	//CHECK: _Ptr<int> e =  (_Ptr<int> ) calloc<int>(1, sizeof(int));
   int *f = (int*) realloc(d, sizeof(int));
-  // CHECK: _Ptr<int> f = (_Ptr<int> ) realloc<int>(d, sizeof(int));
+	//CHECK: _Ptr<int> f =  (_Ptr<int> ) realloc<int>(d, sizeof(int));
 }
 
-// Allocating pointers to pointers
+/* Allocating pointers to pointers */
+	//CHECK: /* Allocating pointers to pointers */
 void bar() {
   int **a = malloc(sizeof(int*));
-  // CHECK: _Ptr<_Ptr<int>> a = malloc<_Ptr<int>>(sizeof(int*));
+	//CHECK: _Ptr<_Ptr<int>> a =  malloc<_Ptr<int>>(sizeof(int*));
   *a = malloc(sizeof(int));
-  // CHECK: *a = malloc<int>(sizeof(int));
+	//CHECK: *a = malloc<int>(sizeof(int));
 
-  // It's also fine if the pointer is unchecked
+  /* It's also fine if the pointer is unchecked */
+	//CHECK: /* It's also fine if the pointer is unchecked */
   int **b = malloc(sizeof(int*));
-  // CHECK: _Ptr<int *> b = malloc<int *>(sizeof(int*));
+	//CHECK: _Ptr<int *> b =  malloc<int *>(sizeof(int*));
   *b = (int*) 1;
+	//CHECK: *b = (int*) 1;
 }
 
-// No conversion is done for void pointers, but this should just test that they
-// convert and compile without crashing. We could insert void as the type
-// parameter if there is anything to be gained from that.
+/* No conversion is done for void pointers, but this should just test that they
+	//CHECK: /* No conversion is done for void pointers, but this should just test that they
```

Comment:
??

Reply:
Hahaha. I experimented a little with auto-adding annotations this time around for some of the files.
The way I add annotations is either if I see some sort of checked type or if I see a base type (int, char, void) and also a pointer (*). This is evidence of how smart it is. Rest assured, I can have this fixed up soon (there's a better version in `testgenerator` that I can use).